File: 2. Add Two Numbers/add_two_number.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
import "fmt"

func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    head := new(ListNode)
    l3 := head
    carry := 0
    for {
        // Initiliase the node
        l3.Val = carry
        carry = 0
        
        // Assign appropriate values
        if l1 != nil {
            l3.Val += l1.Val
            l1 = l1.Next
        }
        if l2 != nil {
            l3.Val += l2.Val
            l2 = l2.Next
        }
        if l3.Val >= 10 {
            carry = l3.Val / 10
            l3.Val = l3.Val % 10
        }
        
        // Iterate or exit appropriately
        if l1 != nil || l2 != nil || carry != 0{
            l3.Next = new(ListNode)
            l3 = l3.Next
        } else {
            l3.Next = nil
            return head
        }
    }
}

File: 2. Add Two Numbers/add_two_number.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        head = None
        prev = None
        carry = 0
        while l1 or l2 or carry == 1:
            newnode = ListNode(0)
            if l1:
                newnode.val += l1.val
                l1 = l1.next
            if l2:
                newnode.val += l2.val
                l2 = l2.next
            newnode.val += carry
            carry = 0
            if newnode.val >= 10:
                carry = 1
                newnode.val %= 10
                
            
            if head:
                prev.next = newnode

            else:
                head = newnode
            prev = newnode

        return head
            
            
